extension/lib/catalog/product/repository/ShopgateImageRepository.js

```diff
@@ -0,0 +1,26 @@
+const ShopgateImageBuilder = require('../service/ShopgateImageBuilder')
+/**
+ * Retrieves images for a product or its variant
+ */
+class ShopgateImageRepository {
+  /**
+   * @param {BigCommerce} apiVersion3Client
+   */
+  constructor (apiVersion3Client) {
+    this._client = apiVersion3Client
+  }
+
+  /**
+   * @param {string} id
+   * @returns {ShopgateImages}
+   */
+  async get (id) {
+    const variantId = id.includes('-') ? Number.parseInt(id.split('-').pop()) : 0
+    const response = await this._client.get('/catalog/products/' + Number.parseInt(id) + '?include=variants,images')
+    const images = new ShopgateImageBuilder(response.data, variantId).build()
```

Comment:
please consider to just return the result of that constructor, instead of saving it in a const, to return that